### ISSUE TITLE
The autocomplete test (73f2c2) needs to be combined with a manual test

### ISSUE THREAD

**Issue by mraccess77** [act-rules/act-rules.github.io#487]
* Use of an invalid autocomplete attribute by itself is not a WCAG violation.
* Use of a valid autocomplete attribute by itself is not a pass of WCAG
* The first manual step of testing SC 1.3.5 is to determine if the field is requesting information about the user.  If the field is not requesting information about the user then whatever autocomplete is there or not there does not fail WCAG.

Recommendation would be to clarify that this test is only tangentially related to WCAG 2.1 and a separate test must be made in conjunction with the current test that can be automated.  Currently the relevance test is a manual test, but could in the future be tested with machine learning while the validity test can be made machine testable.

**Comment by WilcoFiers** [comment]
> Use of an invalid autocomplete attribute by itself is not a WCAG violation.

This rule has an underlying assumption, described here:
https://act-rules.github.io/rules/73f2c2#assumptions

You're correct, if autocomplete is used on a field for which no correct autocomplete value exists, like `<input name="shoe-size" autocomplete="shoe-size" />`, this rule will produce a false positive. My experience with this, and the experience of the reviewers of the rule has been that someone making up an entirely new autocomplete value that serves no purpose is uncommon. Far more likely is that someone will make a mistake when using autocomplete with a field that actually needs it. That is a WCAG 2.1 conformance issue.

There is very little that can be tested automatically, if you need 100% accurate rules. Most tool developers opt for something in the 90% - 99% range. That's a choice each tool developer makes for themselves, depending on their user base. I think that's perfectly fine. My personal take on it is that hypothetical scenarios aren't false positives. If we can't find a real world example of a false positive, that rule is accurate, even if it does make assumptions.

> Use of a valid autocomplete attribute by itself is not a pass of WCAG

ACT-R rules don't "pass" WCAG. This is something that isn't as explicit today as it should be. There is a change, introduced in the candidate rec of the ACT Rules Format which requires that we are explicit about this. A pass of this rule means further testing is still needed for the SC.

> The first manual step of testing SC 1.3.5 is to determine if the field is requesting information about the user. If the field is not requesting information about the user then whatever autocomplete is there or not there does not fail WCAG.

Correct, see my first comment.

**Comment by mraccess77** [comment]
The assumption in my opinion should be extended beyond only fields that allow corresponding autocomplete to include only fields that are asking for data about the current user.  As mentioned and was agreed related to this point the test in terms of goals for SC 1.3.5 are all predicated on the fact the field is asking for data about the current user.  If the field is asking for data about another user -- or about the user's company then SC 1.3.5 doesn't apply and this test is not related to that SC and is more about HTML validation.  This should be added in my opinion as a test step  -- Is this field requesting data about the current user?  If not -- test doesn't apply.

**Comment by WilcoFiers** [comment]
Makes sense. I'll work on that.

**Comment by ajanec01** [comment]
Hi @mraccess77 

The issue you've reaised has been fixed. Feel free to revisit the rule and comment if feel something isn't right.
Here is the link to the rule https://act-rules.github.io/rules/73f2c2

**Comment by mraccess77** [comment]
Thanks for updating this rule.  I don't agree that using an incorrect type fails 1.3.1.  The understanding document for 1.3.5 indicates the type attribute doesn't communicate the purpose of the field but instead communicates the input type.   Thus, use of incorrect type is not a failure because it doesn't change the purpose.

**Comment by ajanec01** [comment]
Hi @mraccess77 
Sorry for the late response. Can you point out the part of the text that says that incorrect `type` attribute fails 1.3.5 (I guess you meant 1.3.5 and not 1.3.1)?

I thought that was made clear in the rule, see the following quote:

> The type attribute is used correctly according to the intended purpose of input elements. If an incorrect type attribute is used for input elements, this rule may fail elements that satisfy success Criterion 1.3.5 Identify Input Purpose. For example if an input element has a type of number, but is expecting an e-mail address.

**Comment by mraccess77** [comment]
Hi, the wording seems to imply that an input with type number that expects an email address may fail SC 1.3.5 but I don't think a misuse of the type attribute would ever fail SC 1.3.5.   The working group's opinion seems to be that the type is not relevant for SC 1.3.5.  I'd like to hear more broadly from the Accessibility Guidelines working group on this.

**Comment by ajanec01** [comment]
Hi,
[Failed example 5](https://act-rules.github.io/rules/73f2c2#failed-example-5) implies that the `input` field has an incorrect programmatic purpose because the purpose is to provide a number and not an email address.  

I agree that the `type` attribute does not need to be used as described in the spec to satisfy 1.3.5 but the edge cases for getting the inappropriate `type` attribute are not too realistic in my opinion. For example, I don't think that the following are common:

```
<input type="password" autocomplete="email" />
```

I have a feeling that the worst case scenario might be something like:

```
 <input type="url" autocomplete="email" />
```

But that also doesn't make much sense. It's the reason why the incorrect `type` attribute for the `autocomplete` value is an assumption made for this rule. I think that it is fair to assume that in the vast majority of cases the `type` attribute will be appropriate.
